include/dev/HallSensor.hpp

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <EVT/io/GPIO.hpp>
+
+namespace IO = EVT::core::IO;
+
+namespace WSS::DEV {
+
+class HallSensor {
+
+public:
+    enum WheelSpeedState {
+        STOP = 0,        // First pulse or wheel is dead
+        INITIALIZING = 1,// Setting speed based on first reading
+        MAINTAIN = 2,    // Wheel is spinning at a constant speed or speeding up
+    };
+
+    // Constructor (take a GPIO instance and the radius of the wheel)
+    HallSensor(IO::GPIO& gpio, uint32_t wheelRadius);
+
+    void update();// Update the wheel speed
+
+    // Get the current speed of the wheel
+    uint32_t getSpeed();
+
+private:
+    IO::GPIO& gpio;       // GPIO instance for the sensor
```

Comment:
Update the format of the instance variable comments:

```suggestion
    /** GPIO instance for the sensor */
    IO::GPIO& gpio;
```

Reply:
Do this for the rest of the variables.